src/main/scala/io/iohk/ethereum/consensus/FullConsensusConfig.scala

```diff
@@ -0,0 +1,31 @@
+package io.iohk.ethereum.consensus
+
+import io.iohk.ethereum.consensus.ethash.MiningConfig
+
+case class FullConsensusConfig[C <: AnyRef /*Product*/](
+  generic: ConsensusConfig,
+  specific: C
+) {
+  // Sanity check
+  require(
+    generic.protocol.isEthash == ifEthash(_ ⇒ true)(false),
+    "generic.protocol.isEthash == ifEthash(_ ⇒ true)(false)"
```

Comment:
It took me a while to understand what this check does. It would be helpful if the message was something like "the specific protocol settings must match the generic ones".
As noted in other comments however, I'm not so fond if `isEthash` and `ifEthash` being implemented the consensus classes themselves.

Reply:
I need to revisit this whole requirement anyway, so thanks for the comment; the message indeed could be more useful in another form.